pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class implementation for state vector measurements.
+"""
+
+# pylint: disable=import-error, no-name-in-module, ungrouped-imports
+try:
+    from pennylane_lightning.lightning_qubit_ops import (
+        MeasurementsC64,
+        MeasurementsC128,
+    )
+except ImportError:
+    pass
+
+from typing import Callable, List
+
+import numpy as np
+from pennylane.measurements import ExpectationMP, MeasurementProcess, StateMeasurement
+from pennylane.tape import QuantumScript
+from pennylane.typing import Result, TensorLike
+from pennylane.wires import Wires
+
+from pennylane_lightning.core._serialize import QuantumScriptSerializer
+
+from ._state_vector import LightningStateVector
+
+
+class LightningMeasurements:
+    """Lightning Measurements class
+
+    Measures the state provided by the LightningStateVector class.
+
+    Args:
+        qubit_state(LightningStateVector)
+    """
+
+    def __init__(self, qubit_state: LightningStateVector) -> None:
+        self._qubit_state = qubit_state
+        self._state = qubit_state.state_vector
+        self._dtype = qubit_state.dtype
+        self._measurement_lightning = self._measurement_dtype()(self.state)
+
+    @property
+    def qubit_state(self):
+        """Returns a handle to the LightningStateVector class."""
+        return self._qubit_state
+
+    @property
+    def state(self):
+        """Returns a handle to the Lightning internal data class."""
+        return self._state
+
+    @property
+    def dtype(self):
+        """Returns the simulation data type."""
+        return self._dtype
+
+    def _measurement_dtype(self):
+        """Binding to Lightning Measurements C++ class.
+
+        Returns: the Measurements class
+        """
+        return MeasurementsC64 if self.dtype == np.complex64 else MeasurementsC128
```

Comment:
Although I agree that a base class would be good to have, I think we can come back to this in the future when we have all classes in place and time allows. Maybe when we start the work to bring the other devices to the new API.

Reply:
Then, it requires removals of more duplicated code! This is easy to maintain now than later. so, I may suggest reconsidering the addition of such base class now  to simplify the implementation of these classes instead of accumulating this technical dept for later ;)

Reply:
Then, it requires removals of more duplicated code! This is easy to maintain now than later, but if details of this base class is not yet clear, we can get back to this later in followup PRs :)